app/main.py

```diff
@@ -1,3 +1,54 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: Union[Distance, float, int]) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[Distance, float, int]) -> Distance:
+        km_of_other = other if not isinstance(other, Distance) else other.km
+        return Distance(
+            self.km + km_of_other
+        )
+
+    def __iadd__(self, other: Union[Distance, float, int]) -> Distance:
+        km_of_other = other if not isinstance(other, Distance) else other.km
+        self.km += km_of_other
+        return self
+
```

Comment:
it's better to use `other.km if isinstance`(remove not condition and simplify if statement). fix everywhere